sympy/series/approximants.py

```diff
@@ -101,3 +108,282 @@ def approximants(l, X=Symbol('x'), simplify=False):
+        f"which is greater than {m}, the requested order of the numerator.")
+
+    approximants = pade_approximants(f, x, m + n)
+
+    numerator, denominator = next(approximants)
+    for next_numerator, next_denominator in approximants:
+        if next_numerator.degree() < m:
+            return numerator, denominator
+
+        numerator, denominator = next_numerator, next_denominator
+
+    if numerator.degree() == m:
+        return numerator, denominator
+
+    # if it hasn't returned yet, there is no [m/n] pade approximant
+    return None, None
+
+
+@public
+def pade_approximant(
+    f: Expr, x: Expr, m: int, n: int | None = None
+    ) -> tuple[Poly, Poly] | None:
+    """
+    `[m/n]` pade approximant of `f` around `x=0`.
+
+    Description
+    ===========
+
+    For a function `f` and integers `m` and `n`, the `[m/n]` pade approximant
+    of `f` is the rational function `p(x)/q(x)`, where `p(x)` and `q(x)` are
+    polynomials of degree at most `m` and `n` respectively. The pade
+    approximation, when it exists, is such that the taylor series of `p(x)/q(x)`
+    around `x=0`matches the taylor series of `f(x)` up to at least order `(m + n)`
+    in `x`.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The function to approximate
+    x : Expr
+        The variable of the function
+    m : int
+        The degree of the numerator polynomial
+    n : int | None
+        The degree of the denominator polynomial. If None, `n = m`
+
+    Returns
+    =======
+
+    pade approximant : tuple[Poly, Poly]
+        returns p(x)
+
+    Examples
+    ========
+
+    >>> import sympy as sp
+    >>> from sympy.series.approximants import pade_approximant
+    >>> x = sp.symbols('x')
+
+    >>> pade_exp = pade_approximant(sp.exp(x), x, 2)
+    >>> pade_exp
+    (x**2/4 + 3*x/2 + 3)/(x**2/4 - 3*x/2 + 3)
+
+    The numerators and denominators of an `[m/n]` pade approximant do not
+    necessarily have order `m` and `n` respectively.
+
+    >>> pade_sin = pade_approximant(sp.sin(x), x, 1, 3)
+    >>> pade_sin
+    36*x/(6*x**2 + 36)
+
+    The `[m/0]` pade approximant is the mth order taylor polynomial of `f`
+
+    >>> pade_cos = pade_approximant(sp.cos(x), x, 4, 0)
+    >>> pade_cos
+    x**4/24 - x**2/2 + 1
+
+    The pade approximant does not always exist, for example, the `[1/1]` approximant for `cos(x)`
+
+    >>> pade_cos11 = pade_approximant(sp.cos(x), x, 1)
+    >>> pade_cos11
+    1
+
+    `1` does not match the taylor expansion of `cos(x)` to second order in `x`.
+
+    See also
+    ========
+
+    sympy.series.approximants.approximants
+    sympy.series.approximants.pade_approximant_gcdex
+    sympy.series.approximants.pade_approximants
+    """
+    if n is None:
+        n = m
+
+    if m < 0 or n < 0:
+        raise ValueError("m and n must be non-negative integers")
+
+    if isinstance(f, Poly):
+        f_taylor_poly = f
+    else:
+        f_taylor_poly = f.series(x, n=m + n + 1).removeO().as_poly(x, field=True)
```

Comment:
I think it would be better to require Expr rather than allowing Poly here as suggested by the type in the signature.